app/controllers/votes_controller.rb

```diff
@@ -2,22 +2,23 @@ class VotesController < ApplicationController
   def create
     @vote = Vote.new(vote_params.merge(user: current_user))
     if @vote.save
-      @count = Vote.where(article_id: params[:article_id]).sum(:value)
-      render json: {id: params[:article_id].to_i, vote: @count, path_up: votes_path(article_id: params[:article_id], value: 1), 
-        path_down: votes_path(article_id: params[:article_id], value: 1)}
+      @count = Vote.where(voteable_id: params[:voteable_id], voteable_type: params[:voteable_type]).sum(:value)
+      
+      render json: {id: params[:voteable_id].to_i, vote: @count, path_up: votes_path(voteable_id: params[:voteable_id], voteable_type: params[:voteable_type],value: 1), 
+        path_down: votes_path(voteable_id: params[:voteable_id], voteable_type: params[:voteable_type], value: -1)}
     end
     
   end
   
   def update
-    @vote = Vote.where(article_id: params[:article_id], user_id: current_user.id).first
+    @vote = Vote.where(voteable_id: params[:voteable_id], voteable_type: params[:voteable_type]).first
     @vote.update_column(:value, params[:value])
```

Comment:
`update_column` bypasses validations, callbacks, and updating the timestamps. Do you intend that?

Also, theoretically, I could go inspect the form that posts to this, see that the value is `1` or `-1`, change that value to `1049982379`, and then now your app's votes are worthless.

I would suggest not putting the `value` into the form, and instead hardcoding that into the controllers, eg, `UpvoteController`, and `DownvoteController`

Reply:
so maybe they're not in the form, but they are in the path that the form POSTs or PUTs to, so it's still editable by the user.